supporting-blog-content/unifying-elastic-vector-database-and-llms-for-intelligent-query/Unifying_Elastic_Vector_Database_and_LLMs_for_Intelligent_Query_Part_2.ipynb

```diff
@@ -0,0 +1,799 @@
+   ],
+   "metadata": {
+    "id": "4jgqRct0C5vG"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "The api key Control security privileges.  The api key security privilages used for this demo:\n",
+    "\n",
+    "```\n",
+    "{\n",
+    "  \"read-only-role\": {\n",
+    "    \"cluster\": [\n",
+    "      \"manage\"\n",
+    "    ],\n",
+    "    \"indices\": [\n",
+    "      {\n",
+    "        \"names\": [\n",
+    "          \"*\"\n",
+    "        ],\n",
+    "        \"privileges\": [\n",
+    "          \"read\"\n",
+    "        ],\n",
+    "        \"allow_restricted_indices\": false\n",
+    "      }\n",
+    "    ],\n",
+    "    \"applications\": [],\n",
+    "    \"run_as\": [],\n",
+    "    \"metadata\": {},\n",
+    "    \"transient_metadata\": {\n",
+    "      \"enabled\": true\n",
+    "    }\n",
+    "  }\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "Caution - This is not how you should design your api key security privilages in your cluster.  This is purely for this demo only."
+   ],
+   "metadata": {
+    "id": "XPaBgE6xzA71"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Install libraries"
+   ],
+   "metadata": {
+    "id": "4lgQ6WlGFeK6"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "kply8eYngIAL"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install \"elasticsearch==8.17.0\" openai streamlit -q"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "wcPv-D6lwHsk"
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import json\n",
+    "from openai import AzureOpenAI\n",
+    "from datetime import datetime, timedelta\n",
+    "from zoneinfo import ZoneInfo\n",
+    "from elasticsearch import Elasticsearch, helpers, NotFoundError\n",
+    "from elasticsearch.helpers import scan, bulk\n",
+    "import requests\n",
+    "from IPython.display import Markdown, display\n",
+    "import requests, json\n",
+    "from getpass import getpass\n",
+    "from google.colab import userdata\n",
```

Comment:
Can you mention at the top of the document that this notebook must run in a Google Colab environment?